test/train/test_train_utils.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+from fruit_classifier.train.train_utils import get_image_paths
+from fruit_classifier.train.train_utils import get_data_and_labels
+from fruit_classifier.train.train_utils import get_model_input
+from fruit_classifier.train.train_utils import get_model
+from fruit_classifier.train.train_utils import train_model
+from fruit_classifier.train.train_utils import plot_training
+from fruit_classifier.preprocessing.preprocessing_utils import \
+    get_image_generator
+from pathlib import Path
+import shutil
+
+
+class TestTrainUtils(unittest.TestCase):
+    def setUp(self):
+        test_dir = Path(__file__).absolute().parents[1]
+
+        # Select a unique folder name for a new folder
+        folder_name = 'temp_folder'
+        self.folder_name = test_dir.joinpath(folder_name)
+        while self.folder_name.is_dir():
+            folder_name = folder_name + '_'
+            self.folder_name = test_dir.joinpath(folder_name)
+
+        # Make directory and two subdirectories
+        self.folder_name.mkdir(parents=True, exist_ok=True)
+        self.folder_name.joinpath('A').mkdir(parents=True, exist_ok=True)
+        self.folder_name.joinpath('B').mkdir(parents=True, exist_ok=True)
+
+        # Copy an image into each of those subdirectories
+        orig_file_path = test_dir.joinpath("test_data").\
+            joinpath("original_test_image.jpg")
+        shutil.copy(str(orig_file_path),
+                    str(self.folder_name.joinpath('A')))
+        shutil.copy(str(orig_file_path),
+                    str(self.folder_name.joinpath('B')))
+
+    def test_train_model(self):
+        # Run get_image_paths() and verify outputs
+        image_paths = get_image_paths(self.folder_name)
+        num_image_paths = len(image_paths)
+        self.assertGreater(num_image_paths, 0)
+
+        # Run get_data_and_labels and verify outputs
```

Comment:
I see that I could have explained better.
I liked the way that you divided in functions previously, my only concern was that `setUp` was full of unrelated stuff